snapcraft/internal/pluginhandler.py

```diff
@@ -874,3 +881,23 @@ def check_for_collisions(parts):
         # And add our files to the list
         parts_files[part.name] = {'files': part_files,
                                   'installdir': part.installdir}
+
+
+def _combine_filesets(fileset_1, fileset_2):
```

Comment:
@sergiusens What are your thoughts on this?  I think @kyrofa has a good point and the logic shouldn't be too complex.